egui/src/ui.rs

```diff
@@ -1631,6 +1637,28 @@ impl Ui {
         self.advance_cursor_after_rect(Rect::from_min_size(top_left, size));
         result
     }
+    /// Close menu (with submenus), if any.
+    pub fn close(&mut self) {
+        if let Some(menu_state) = &mut self.menu_state {
+            menu_state.write().unwrap().close();
+        }
+        self.menu_state = None;
+    }
+    pub(crate) fn set_menu_state(&mut self, menu_state: Arc<RwLock<MenuState>>) {
+        self.menu_state = Some(menu_state);
+    }
+    #[inline(always)]
+    pub fn menu<R>(
```

Comment:
This needs a docstring and a small example!

Also I think `menu_button` is a better name.